From 72089ac06f71d4b790a175bca0e47be4aa97792d Mon Sep 17 00:00:00 2001
From: Debbie Martin <Debbie.Martin@arm.com>
Date: Tue, 5 Sep 2023 21:26:52 +0100
Subject: [PATCH] check-sr-results: Change the expected SR result config

Update the check-sr-results.yaml and dt-parser.yaml files for the Kronos
Systemready IR suite. These changes are required because of the
following known differences of fvp-rd-kronos outputs to those expected
by the SystemReady scripts.

Changes to check-sr-results.yaml:

  1. acs-console.log:
      a. must-have-esp: EFI partition is not fully supported.
  2. acs_results/result.md:
      a. must-contain "Failure: |3|": UpdateCapsule failure make the
         total 3.
  3. acs_results/fwts/FWTSResults.log:
      a. Update SystemReady IR ACS version to v2.1.0.
      b. Running tests: Update the test case list.
      c. Add fwts test summary.
  4. acs_results/linux_acs/bsa_acs_app/BSALinuxResults.log:
      a. Update SystemReady IR ACS version to v2.1.0.
  5. acs_results/linux_dump/firmware/efi/efivars:
      a. Make the min-entries 0: EFI partition is not fully supported.
  6. acs_results/linux_dump/uname.log:
      a. Update Linux version to 6.4.0
  7. acs_results/linux_tools:
      a. Set min-entries to 3: Currently device_driver_info.log,
         device_tree.dts and dt-validate.log are available,
         ethtool-test.log and read_blk_devices.log are sometimes
         unavailable. Further analysis is required.
  8. acs_results/linux_tools/ethtool-test.log:
      a. Make optional because the log file is sometimes unavailable and
         further analysis is required.
  9. acs_results/linux_tools/read_blk_devices.log:
      a. Make optional because the log file is sometimes unavailable and
         further analysis is required.
  10. acs_results/sct_results:
      a. Update SystemReady IR ACS version to v2.1.0.
  11. acs_results/uefi/BsaDevTree.dtb:
      a. Add extra_compat list for the devicetree.
  12. acs_results/uefi/temp:
      a. Set min-entries to 0: this defaults to 1 despite the directory
         being optional, so errors if a directory is empty.
  13. acs_results/uefi_dump:
      a. min-entries is 13: change this to 11 due to smbiosview.log
         being optional and the change to make map.log optional.
  14. acs_results/uefi_dump/devtree.log:
      a. must-contain: there's no "/HD(3,GPT,)" in IR ACS v2.1.0.
  15. acs_results/uefi_dump/dh.log:
      a. must-contain "EFISystemPartition": EFI partition is not fully
         supported.
  16. acs_results/uefi_dump/dmpstore.log:
      a. must-contain "Variable": EFI partition is not fully supported.
      b. must-contain "SecureBoot": SecureBoot is not supported.
  17. acs_results/uefi_dump/map.log:
      a. Make optional because it isn't populated for IR.
  18. fw:
      a. Make optional because capsule updates are not fully supported.
  19. os-logs:
      a. Make optional because distro installation isn't done as part
         of ACS.
  20. os-logs/linux-*:
      a. Set min-occurrences to 0 because distro installation isn't done
         as part of ACS.
  21. report.txt:
      a. must-contain: Update the strings to match IR ACS v2.1.0
      b. warn-if-contains "[ ]": Capsule updates are not fully
         supported.
  22. acs_results/sct_results/Overall/Summary.log:
      a. Update SystemReady IR ACS version to v2.1.0.

Changes to dt-parser.yaml:

  1. Ignore missing schema warnings for the added extra compat entries.

Upstream-Status: Inappropriate
Signed-off-by: Debbie Martin <Debbie.Martin@arm.com>
Signed-off-by: Robbie Cao <robbie.cao@arm.com>

%% original patch: 0001-check-sr-results-Change-the-expected-SR-result-confi.patch
---
 check-sr-results.yaml | 61 +++++++++++++++++++++++++++----------------
 dt-parser.yaml        | 48 ++++++++++++++++++++++++++++++++++
 2 files changed, 86 insertions(+), 23 deletions(-)

diff --git a/check-sr-results.yaml b/check-sr-results.yaml
index fe5aec1..f81a745 100644
--- a/check-sr-results.yaml
+++ b/check-sr-results.yaml
@@ -10,7 +10,6 @@ check-sr-results-configuration:
 # The following tree applies to all ACS-IR 2.0 versions.
 tree:
   - file: acs-console.log
-    must-have-esp:
     must-contain:
       - Booting `bbr/bsa'
       - Press any key to stop the EFI SCT running
@@ -53,7 +52,7 @@ tree:
         must-contain:
           - SCT Summary
           - Dropped:|0|
-          - Failure:|0|
+          - Failure:|3|
           - Warning:|0|
           - Dropped by group
           - Failure by group
@@ -95,11 +94,11 @@ tree:
         tree:
           - file: FWTSResults.log
             must-contain:
-              - SystemReady IR ACS v2.0.0
+              - SystemReady IR ACS v2.1.0
               - FWTS v2
               - 'Results generated by fwts: Version V2'
-              - 'Running tests: uefivarinfo esrt uefibootpath dt_base uefirtmisc
-                uefirtvariable'
+              - 'Running tests: uefivarinfo esrt uefibootpath dt_base dt_base esrt uefibootpath'
+              - 'uefivarinfo uefirtmisc uefirtvariable uefirttime'
               - 'esrt: Sanity check UEFI ESRT Table.'
               - Test Failure Summary
               - 'Critical failures: NONE'
@@ -107,6 +106,13 @@ tree:
               - 'Medium failures: NONE'
               - 'Low failures: NONE'
               - 'Other failures: NONE'
+              - 'dt_base        |    3|     |     |     |     |     |'
+              - 'esrt           |    2|     |     |     |     |     |'
+              - 'uefibootpath   |     |     |     |     |     |     |'
+              - 'uefirtmisc     |    1|     |     |     |    8|     |'
+              - 'uefirttime     |    4|     |     |     |   35|     |'
+              - 'uefirtvariable |    2|     |     |     |   10|     |'
+              - 'uefivarinfo    |     |     |     |     |    1|     |'
               - 'Total:'
             warn-if-contains:
               - Aborted test
@@ -130,7 +136,7 @@ tree:
             tree:
               - file: BSALinuxResults.log
                 must-contain:
-                  - SystemReady IR ACS v2.0.0
+                  - SystemReady IR ACS v2.1.0
                   - BSA v1.0.
                   - BSA Architecture Compliance Suite
                   - Version 1.0.
@@ -189,6 +195,7 @@ tree:
                 min-entries: 1
                 tree:
                   - dir: efivars
+                    min-entries: 0
                     tree:
                       - file: OsIndicationsSupported-*
                   - dir: esrt
@@ -245,10 +252,10 @@ tree:
               - block
           - file: uname.log
             must-contain:
-              - 'Linux generic-arm64 6.0.0-yocto-standard #1 SMP PREEMPT Sun Oct
-                2 21:09:07 UTC 2022 aarch64 GNU/Linux'
+              - 'Linux generic-arm64 6.4.0-yocto-standard+ #1 SMP PREEMPT
+                 Sun Jun 25 23:29:58 UTC 2023 aarch64 GNU/Linux'
       - dir: linux_tools
-        min-entries: 4
+        min-entries: 3
         max-entries: 6  # Allow dt-validate.log.old
         tree:
           - file: device_tree.dts
@@ -261,9 +268,11 @@ tree:
           - file: device_driver_info.log    #This log may be removed in  future versions
             optional:
           - file: ethtool-test.log
+            optional:
             must-contain:
               - 'INFO: Detected following ethernet interfaces via ip command :'
           - file: read_blk_devices.log
+            optional:
             must-contain:
               - 'INFO: Detected following block devices with lsblk command :'
       - dir: sct_results
@@ -288,7 +297,7 @@ tree:
                   - '|TERM|'
               - file: Summary.log
                 must-contain:
-                  - 'Arm ACS Version: SystemReady IR ACS v2.0.0'
+                  - 'Arm ACS Version: SystemReady IR ACS v2.1.0'
                   - BBR ACS 1.0.
                   - 'Test Finished:'
           - dir: Sequence
@@ -325,6 +334,15 @@ tree:
         tree:
           - file: BsaDevTree.dtb
             devicetree:
+            extra_compat:
+              - arm,rd-kronos
+              - arm,neoverse
+              - arm,slc
+              - cache
+              - arm,mpam-msc
+              - arm,mhuv3
+              - arm,si-rproc
+              - arm,si-channel
           - file: BsaDevTree.dtb.log  # This will be generated
             must-contain:
               - + DTC
@@ -342,8 +360,9 @@ tree:
           - dir: temp   # This sometimes remains; ignore it
             min-entries: 0
             optional:
+            min-entries: 0
       - dir: uefi_dump
-        min-entries: 13
+        min-entries: 11
         max-entries: 13
         tree:
           - file: bcfg.log
@@ -354,12 +373,10 @@ tree:
             must-contain:
               - /HD(1,GPT,
               - /HD(2,GPT,
-              - /HD(3,GPT,
           - file: dh.log
             must-contain:
               - Handle dump
               - DevicePath
-              - EFISystemPartition
               - FirmwareManagement
               - SimpleTextOut
           - file: dmem.log
@@ -375,14 +392,12 @@ tree:
               - MPS Table
               - SMBIOS Table
           - file: dmpstore.log
-            must-contain:
-              - Variable
-              - SecureBoot
           - file: drivers.log
             must-contain:
               - DRIVER NAME
           - file: ifconfig.log
           - file: map.log
+            optional:
             must-contain:
               - Mapping table
               - /HD(1,GPT,
@@ -421,7 +436,7 @@ tree:
     optional:
     min-entries: 0
   - dir: fw
-    min-entries: 3
+    optional:
     tree:
       - file: u-boot-sniff.log
         must-contain:
@@ -529,14 +544,14 @@ tree:
   - dir: manual-results
     optional:
   - dir: os-logs
-    min-entries: 2
+    optional:
     tree:
       - file: 'OS-image-download-links.txt'
         optional:
         must-contain:
           - For SystemReady IR
       - dir: 'linux-*'
-        min-occurrences: 2
+        min-occurrences: 0
         tree:
           - file: '*console.log'
             warn-if-not-named: console.log
@@ -563,9 +578,9 @@ tree:
   - file: report.txt
     must-contain:
       - General information
-      - Test logs and results collected
-    warn-if-contains:
-      - '[ ]'
+      - System information
+      - Test Logs and Results
+      - Test Logs collected
 
 overlays:
 
@@ -668,7 +683,7 @@ overlays:
                           - '|TERM|'
                       - file: Summary.log
                         must-contain:
-                          - 'Arm ACS Version: SystemReady IR ACS v2.0.0'
+                          - 'Arm ACS Version: SystemReady IR ACS v2.1.0'
                           - BBR ACS 1.0.
                           - 'Test Finished:'
                   - dir: Sequence
diff --git a/dt-parser.yaml b/dt-parser.yaml
index dc62310..0717953 100644
--- a/dt-parser.yaml
+++ b/dt-parser.yaml
@@ -198,6 +198,54 @@
 # We transform the following dt-validate warning into errors as those issues
 # will block certification.
 
+- rule: Allow no schema for arm,si-channel
+  criteria:
+    type: dt-validate warning
+    warning_message: 'failed to match any schema with compatible: '
+    in_compatibles: 'arm,si-channel'
+  update:
+    type: ignored
+
+- rule: Allow no schema for arm,mpam-msc
+  criteria:
+    type: dt-validate warning
+    warning_message: 'failed to match any schema with compatible: '
+    in_compatibles: 'arm,mpam-msc'
+  update:
+    type: ignored
+
+- rule: Allow no schema for arm,mhuv3
+  criteria:
+    type: dt-validate warning
+    warning_message: 'failed to match any schema with compatible: '
+    in_compatibles: 'arm,mhuv3'
+  update:
+    type: ignored
+
+- rule: Allow no schema for arm,rd-kronos, arm,neoverse
+  criteria:
+    type: dt-validate warning
+    warning_message: 'failed to match any schema with compatible: '
+    in_compatibles: 'arm,rd-kronos, arm,neoverse'
+  update:
+    type: ignored
+
+- rule: Allow no schema for arm,slc, cache
+  criteria:
+    type: dt-validate warning
+    warning_message: 'failed to match any schema with compatible: '
+    in_compatibles: 'arm,slc, cache'
+  update:
+    type: ignored
+
+- rule: Allow no schema for arm,si-rproc
+  criteria:
+    type: dt-validate warning
+    warning_message: 'failed to match any schema with compatible: '
+    in_compatibles: 'arm,si-rproc'
+  update:
+    type: ignored
+
 - rule: Categorize all dt-validate warnings about missing schema and with no
     matching compatible as `no schema' errors, which will block certification.
   criteria:
-- 
2.25.1

